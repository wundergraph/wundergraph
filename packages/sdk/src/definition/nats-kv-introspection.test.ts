import { z } from 'zod';
import { introspectNatsKV } from './nats-kv-introspection';

describe('nats kv', () => {
	describe('introspection', function () {
		it('should work for simple models', async function () {
			const introspection = await introspectNatsKV({
				apiNamespace: 'token',
				model: z.object({
					token: z.string(),
				}),
			});
			const out = await introspection({});
			expect(out.Schema).toMatchSnapshot();
			out.Schema = '';
			expect(JSON.stringify(out, null, '  ')).toMatchSnapshot();
		});
		it('should work for nested models', async function () {
			const introspection = await introspectNatsKV({
				apiNamespace: 'token',
				model: z.object({
					token: z.string(),
					user: z.object({
						id: z.number(),
					}),
					org: z.object({
						id: z.number(),
					}),
				}),
			});
			const out = await introspection({});
			expect(out.Schema).toMatchSnapshot();
			out.Schema = '';
			expect(JSON.stringify(out, null, '  ')).toMatchSnapshot();
		});
	});
});
