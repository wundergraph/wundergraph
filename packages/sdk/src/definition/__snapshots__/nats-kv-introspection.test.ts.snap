// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nats kv introspection should work for nested models 1`] = `
"schema {
  query: Query
  mutation: Mutation
  subscription: Subscription
}

type Query {
  "Get returns the latest value for the key."
  token_get(key: String!): token_KeyValueEntry
  "GetRevision returns a specific revision value for the key."
  token_getRevision(key: String!, revision: Int!): token_KeyValueEntry
  "Keys will return all keys."
  token_keys: [String!]!
  "History will return all historical values for the key."
  token_history(key: String!): [token_KeyValueEntry!]!
}

type Mutation {
  "Put will place the new value for the key into the store."
  token_put(key: String!, value: token_InputValue!): token_KeyValueEntry
  "Create will add the key/value pair iff it does not exist."
  token_create(key: String!, value: token_InputValue!): token_KeyValueEntry
  "Update will update the value iff the latest revision matches."
  token_update(key: String!, value: token_InputValue!, revision: Int!): token_KeyValueEntry
  "Delete will place a delete marker and leave all revisions."
  token_delete(key: String!): Boolean!
  "Purge will place a delete marker and remove all previous revisions."
  token_purge(key: String!): Boolean!
}

type Subscription {
  """
   Watch for any updates to keys that match the keys argument which could include wildcards.
  Watch will send a nil entry when it has received all initial values.
  """
  token_watch(keys: [String!]!): [token_KeyValueEntry!]!
  "WatchAll will invoke the callback for all updates."
  token_watchAll: [token_KeyValueEntry!]!
}

type token_KeyValueEntry {
  key: String!
  value: token_Value
  revision: Int!
  created: Int!
}

input token_InputValueUser {
  id: Float!
}

input token_InputValueOrg {
  id: Float!
}

input token_InputValue {
  token: String!
  user: token_InputValueUser!
  org: token_InputValueOrg!
}

type token_ValueUser {
  id: Float!
}

type token_ValueOrg {
  id: Float!
}

type token_Value {
  token: String!
  user: token_ValueUser!
  org: token_ValueOrg!
}"
`;

exports[`nats kv introspection should work for nested models 2`] = `
"{
  "DefaultFlushInterval": 500,
  "Schema": "",
  "Namespace": "token",
  "DataSources": [
    {
      "RootNodes": [
        {
          "typeName": "Query",
          "fieldNames": [
            "token_get"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 0,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Query",
          "fieldNames": [
            "token_getRevision"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 1,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Query",
          "fieldNames": [
            "token_keys"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 2,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Query",
          "fieldNames": [
            "token_history"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 3,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Mutation",
          "fieldNames": [
            "token_put"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 4,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Mutation",
          "fieldNames": [
            "token_create"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 5,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Mutation",
          "fieldNames": [
            "token_update"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 6,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Mutation",
          "fieldNames": [
            "token_delete"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 7,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Mutation",
          "fieldNames": [
            "token_purge"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 8,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Subscription",
          "fieldNames": [
            "token_watch"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 9,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Subscription",
          "fieldNames": [
            "token_watchAll"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 10,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            },
            "user": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            },
            "org": {
              "type": "object",
              "properties": {
                "id": {
                  "type": "number"
                }
              },
              "required": [
                "id"
              ],
              "additionalProperties": false
            }
          },
          "required": [
            "token",
            "user",
            "org"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    }
  ],
  "Fields": [
    {
      "typeName": "Query",
      "fieldName": "token_get",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Query",
      "fieldName": "token_getRevision",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Query",
      "fieldName": "token_keys",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Query",
      "fieldName": "token_history",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Mutation",
      "fieldName": "token_put",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Mutation",
      "fieldName": "token_create",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Mutation",
      "fieldName": "token_update",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Mutation",
      "fieldName": "token_delete",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Mutation",
      "fieldName": "token_purge",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Subscription",
      "fieldName": "token_watch",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Subscription",
      "fieldName": "token_watchAll",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    }
  ],
  "Types": [],
  "interpolateVariableDefinitionAsJSON": [],
  "CustomJsonScalars": {}
}"
`;

exports[`nats kv introspection should work for simple models 1`] = `
"schema {
  query: Query
  mutation: Mutation
  subscription: Subscription
}

type Query {
  "Get returns the latest value for the key."
  token_get(key: String!): token_KeyValueEntry
  "GetRevision returns a specific revision value for the key."
  token_getRevision(key: String!, revision: Int!): token_KeyValueEntry
  "Keys will return all keys."
  token_keys: [String!]!
  "History will return all historical values for the key."
  token_history(key: String!): [token_KeyValueEntry!]!
}

type Mutation {
  "Put will place the new value for the key into the store."
  token_put(key: String!, value: token_InputValue!): token_KeyValueEntry
  "Create will add the key/value pair iff it does not exist."
  token_create(key: String!, value: token_InputValue!): token_KeyValueEntry
  "Update will update the value iff the latest revision matches."
  token_update(key: String!, value: token_InputValue!, revision: Int!): token_KeyValueEntry
  "Delete will place a delete marker and leave all revisions."
  token_delete(key: String!): Boolean!
  "Purge will place a delete marker and remove all previous revisions."
  token_purge(key: String!): Boolean!
}

type Subscription {
  """
   Watch for any updates to keys that match the keys argument which could include wildcards.
  Watch will send a nil entry when it has received all initial values.
  """
  token_watch(keys: [String!]!): [token_KeyValueEntry!]!
  "WatchAll will invoke the callback for all updates."
  token_watchAll: [token_KeyValueEntry!]!
}

type token_KeyValueEntry {
  key: String!
  value: token_Value
  revision: Int!
  created: Int!
}

input token_InputValue {
  token: String!
}

type token_Value {
  token: String!
}"
`;

exports[`nats kv introspection should work for simple models 2`] = `
"{
  "DefaultFlushInterval": 500,
  "Schema": "",
  "Namespace": "token",
  "DataSources": [
    {
      "RootNodes": [
        {
          "typeName": "Query",
          "fieldNames": [
            "token_get"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 0,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Query",
          "fieldNames": [
            "token_getRevision"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 1,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Query",
          "fieldNames": [
            "token_keys"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 2,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Query",
          "fieldNames": [
            "token_history"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 3,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Mutation",
          "fieldNames": [
            "token_put"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 4,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Mutation",
          "fieldNames": [
            "token_create"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 5,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Mutation",
          "fieldNames": [
            "token_update"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 6,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Mutation",
          "fieldNames": [
            "token_delete"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 7,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Mutation",
          "fieldNames": [
            "token_purge"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 8,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Subscription",
          "fieldNames": [
            "token_watch"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 9,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    },
    {
      "RootNodes": [
        {
          "typeName": "Subscription",
          "fieldNames": [
            "token_watchAll"
          ]
        }
      ],
      "ChildNodes": [],
      "Directives": [],
      "Kind": 9,
      "Custom": {
        "bucketPrefix": {
          "kind": 1,
          "environmentVariableName": "WG_NATS_PREFIX",
          "environmentVariableDefaultValue": "",
          "placeholderVariableName": "",
          "staticVariableContent": ""
        },
        "serverURL": "",
        "token": "",
        "history": 1,
        "bucketName": "token",
        "operation": 10,
        "schema": {
          "type": "object",
          "properties": {
            "token": {
              "type": "string"
            }
          },
          "required": [
            "token"
          ],
          "additionalProperties": false,
          "$schema": "http://json-schema.org/draft-07/schema#"
        }
      },
      "RequestTimeoutSeconds": 10
    }
  ],
  "Fields": [
    {
      "typeName": "Query",
      "fieldName": "token_get",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Query",
      "fieldName": "token_getRevision",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Query",
      "fieldName": "token_keys",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Query",
      "fieldName": "token_history",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Mutation",
      "fieldName": "token_put",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Mutation",
      "fieldName": "token_create",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Mutation",
      "fieldName": "token_update",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Mutation",
      "fieldName": "token_delete",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Mutation",
      "fieldName": "token_purge",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Subscription",
      "fieldName": "token_watch",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    },
    {
      "typeName": "Subscription",
      "fieldName": "token_watchAll",
      "disableDefaultFieldMapping": true,
      "unescapeResponseJson": false,
      "requiresFields": [],
      "path": [],
      "argumentsConfiguration": []
    }
  ],
  "Types": [],
  "interpolateVariableDefinitionAsJSON": [],
  "CustomJsonScalars": {}
}"
`;
